torch/ao/quantization/pt2e/lowering.py

```diff
@@ -0,0 +1,60 @@
+import torch
+from torch._inductor.constant_folding import constant_fold
+from torch._inductor.fx_passes.freezing_patterns import freezing_passes
+
+
+__all__ = [
+    "lower_pt2e_quantized_to_x86",
+]
+
+
+def lower_pt2e_quantized_to_x86(
+    model: torch.fx.GraphModule,
+    example_inputs: tuple[torch.Tensor, ...],
+) -> torch.fx.GraphModule:
+    """Lower a PT2E-qantized model to x86 backend.
+
+    Args:
+    * `model` (torch.fx.GraphModule): a model quantized by PT2E quantization flow.
+    * `example_inputs` (tuple[torch.Tensor, ...]): example inputs for the model.
+
+    Return:
+    A GraphModule lowered to x86 backend.
+    """
+
+    def _post_autograd_decomp_table():  # type: ignore[no-untyped-def]
+        decomp_table = torch.export.default_decompositions()
+
+        # if we are post-autograd, we shouldn't
+        # decomp prim ops.
+        for k in list(decomp_table.keys()):
+            if not torch._export.utils._is_cia_op(k):
+                del decomp_table[k]
+
+        return decomp_table
+
+    def _node_replace(m):  # type: ignore[no-untyped-def]
+        # Replace aten.t(x) with aten.permute(x, [1, 0])
+        aten = torch.ops.aten
+        g = m.graph
+        for node in g.nodes:
+            if node.target == aten.t.default:
+                with g.inserting_before(node):
+                    x = node.args[0]
+                    dims = [1, 0]
+                    perm_node = g.call_function(aten.permute.default, args=(x, dims))
+                    node.replace_all_uses_with(perm_node)
+                    g.erase_node(node)
+
+        g.lint()
+        m.recompile()
+
+    lowered_model = (
+        torch.export.export_for_training(model, example_inputs)
```

Comment:
Just wondering since the model after `pt2e` is already be `export_for_training`, is it a must have to do it again here? And is there any requirement for the input model of this API `lower_pt2e_quantized_to_x86`?

Reply:
It is exported again to call the `run_decompositions()`. The input model should be the output of `convert_pt2e`. Thanks.

Reply:
Can we add an assert to check the input model is as we expected?

Reply:
Could you let me know how to check if a module is from `convert_pt2e`? Thanks.

Reply:
Is the concern about extra time required to check the `quant/dequant` nodes in the input Graph Module? TBH, I don't have a better approach than this.

Reply:
Checking quant/dequant does not seem good. What if the module has quant/dequant but the pattern is not what we expected? Anyway, I don't think we have a strong motivation to check the graph here because
1. Users are expected to follow the workflow. `convert_pt2e` does not check its input is from `prepare_pt2e` eiter.
2. Even the original model (non-pt2e-quantized) can be processed by this API and the returned model is a non-quantized lowered model.

Do you have any concern? Thanks.

Reply:
Another reason: the lowering is all about decomposition and pattern matching. There won't be any side effect if patterns are not matched.